numpy/random/bit_generator.pxd

```diff
@@ -11,7 +11,15 @@ cdef class BitGenerator():
     cdef readonly object capsule
 
 
-cdef class SeedSequence():
+cdef class ISeedSequence():
+    pass
```

Comment:
I recommend using an `ABC` as in my gist with the `@abc.abstractmethod` definition of `generate_state()`. It'll be a plain `class` rather than a `cdef class` because Cython restricts things, so you won't subclass `SeedSequence` from it, just `ISeedSequence.register(SeedSequence)` after both are defined. Please make `SeedlessSeedSequence` actually implement the interface as I described.

Reply:
We are running up against restrictions in cython which does not seem to support ABC. Can you get the code to compile as you desire?

Reply:
Did you try what I described? I'll give it a shot myself once I get my git checkout in order.

Reply:
Yes, [it works fine](https://github.com/mattip/numpy/pull/38).

Reply:
ABC fixed by @rkern